app/lib/attachment_batch.rb

```diff
@@ -78,9 +78,17 @@ def remove_files
             logger.debug { "Deleting #{attachment.path(style)}" }
 
             begin
+              retries ||= 0
               attachment.send(:directory).files.new(key: attachment.path(style)).destroy
-            rescue Fog::Storage::OpenStack::NotFound
-              # Ignore failure to delete a file that has already been deleted
+            rescue => e
+              if e.is_a?(Fog::OpenStack::Storage::NotFound)
```

Comment:
There is actually no need for this, as you can have more than one `rescue` clause handling different types of exceptions. So this can be rewritten as

```ruby
begin
  attachment.send(:directory).files.new(key: attachment.path(style)).destroy
rescue Fog::OpenStack::Storage::NotFound
  logger.debug "Will ignore because file is not found #{attachment.path(style)}"
rescue => e
  sleep(5)
  retry if (retries += 1) < 3
  logger.error "Batch deletion from fog failed after #{e.message}"
  raise e
end  
```

Reply:
This is the solution I found before `rescue Fog::Storage::OpenStack::NotFound` crashes with "uninitialized constant Fog::Storage::OpenStack (NameError)" but `if e.is_a?(Fog::OpenStack::Storage::NotFound)` does not.

My understanding here is minimal and I agree that having the NotFound on its own rescue would be best, I just did not find a way to make it work. Any suggestions would be appreciated.

Reply:
It seems the correct class is `Fog::OpenStack::Storage::NotFound`. I don't know if it has changed recently or has always been wrong, going to investigate.

Reply:
It did change with a `fog-openstack` update: https://github.com/fog/fog-openstack/commit/391cda23afad6c0aa1ccd878e71250419a727edc

Reply:
Perfect! I changed the rescue to use the new `Fog::OpenStack::Storage::NotFound` and successfully tested it on my end. 

And sorry @oneiros I did not notice that you had suggested the use of the new class (`Fog::OpenStack` vs `Fog::Storage`) when I read your code suggestion above but your suggest was correct. Thanks.